### PR TITLE
Refactor, settings, colored spectrograms, UI changes

### PR DESCRIPTION
**Pull request by richardkunkli**
# Fixes
- Refactor BoringViewModel observed object to initialize in the appdelegate. It is important because the new settings changes variables of the env object and it has to be in the same instance
- Changed icon in the HeaderExtras, because it was confusing
# New features
- Implemented a new extension that extract the average color from the album art, and it is used in the audio spectrogram, which has been optimized and resized
- New settings window with some options, giving users ability to customize their experience
# Preview
Screenshot 2024-08-07 at 20 41 51

### CODE REVIEW

**Comment by iamharshdev** [comment]
Thanks for @richardkunkli this but please make sure to add some preview here 🔥

**Comment by richardkunkli** [comment]
Noted. Added a preview.

**Comment by iamharshdev** [comment]
🔥 Loved the changes!

File: boringNotch/components/BoringNotch.swift [hunk 1]
```diff
@@ -99,25 +101,21 @@ struct BoringNotch: View {
...
-                    
                     if vm.notchState == .closed && batteryModel.showChargingInfo {
                         BoringBatteryView(batteryPercentage: batteryModel.batteryPercentage, isPluggedIn: batteryModel.isPluggedIn)
                     }
                     
-                    if musicManager.isPlaying && !batteryModel.showChargingInfo && vm.currentView != .menu {
+                    if !batteryModel.showChargingInfo && vm.currentView != .menu {
                         
-                        MusicVisualizer()
+                        MusicVisualizer(avgColor: musicManager.albumArt.averageColor(), isPlaying: musicManager.isPlaying)
```
Comment (iamharshdev) [inline boringNotch/components/BoringNotch.swift:110]: You should never do extensive tasks on application thread. @richardkunkli 
I'll fix this no worries.

**Pull request merged**